drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,469 @@
+			poll_again = false;
+		}
+
+		/* Subtract the time invested from the sleep period */
+		current_cycles = k_cycle_get_32();
+		cycles_diff = current_cycles - start_period_cycles;
+		wait_period = data->poll_period_us - k_cyc_to_us_floor32(cycles_diff);
+
+		/* Override wait_period in case it is less than 1 ms */
+		if (wait_period < USEC_PER_MSEC) {
+			wait_period = USEC_PER_MSEC;
+		}
+
+		if (wait_period > data->poll_period_us) {
+			wait_period = data->poll_period_us;
+		}
+
+		if (poll_again) {
+			k_work_schedule(&data->polling_work, K_USEC(wait_period));
+		}
+	}
+}
+
+static void interrupt_handler(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	LOG_DBG("stmpe811 generated an interrupt");
+	struct stmpe811_data *data = CONTAINER_OF(cb, struct stmpe811_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static int stmpe811_verify_chip_id(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	uint8_t buf[2];
+
+	i2c_burst_read_dt(&config->bus, STMPE811_CHP_ID_LSB_REG, buf, 2);
+
+	if (((buf[0] << 8) | buf[1]) != CHIP_ID) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stmpe811_init(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	if (!device_is_ready(config->bus.bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	k_work_init(&data->work, stmpe811_work_handler);
+
+	/* Verify CHIP_ID */
+	if (stmpe811_verify_chip_id(dev)) {
+		LOG_ERR("CHIP ID verification failed");
+		return -EINVAL;
+	}
+
+	/* Initialize and enable interrupts */
+	int err = stmpe811_ts_init(dev);
+
+	if (err) {
+		LOG_ERR("Touch screen controller initialization failed");
+		return err;
+	}
+
+	err = stmpe811_ts_enable_it(dev);
+	if (err) {
+		LOG_ERR("Could not enable touch screen interrupts");
+		return err;
+	}
+
+	/* Initialize GPIO interrupt */
+	if (!device_is_ready(config->int_gpio.port)) {
+		LOG_ERR("Interrupt GPIO controller device not ready");
+		return -ENODEV;
+	}
+
+	err = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+
+	if (err < 0) {
+		LOG_ERR("Could not configure interrupt GPIO pin");
+		return err;
+	}
+
+	err = gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+	if (err < 0) {
+		LOG_ERR("Could not configure GPIO interrupt");
+		return err;
+	}
+
+	gpio_init_callback(&data->int_gpio_cb, interrupt_handler, BIT(config->int_gpio.pin));
+	err = gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+	if (err < 0) {
+		LOG_ERR("Could not set GPIO callback");
+		return err;
+	}
+
+	/* Initialize the release polling */
+	k_work_init_delayable(&data->polling_work, stmpe811_poll_release_work_handler);
+	data->poll_timeout_us = config->hold_timeout * USEC_PER_MSEC;
+	data->poll_period_us = config->polling_period * USEC_PER_MSEC;
+	data->poll_release_threshold = config->release_counter_threshold;
```

Comment:
Same, just use the config version directly.

Reply:
Done